hc/api/tests/test_create_check.py

```diff
@@ -44,26 +46,35 @@ def test_it_works(self):
         self.assertEqual(check.tags, "bar,baz")
         self.assertEqual(check.timeout.total_seconds(), 3600)
         self.assertEqual(check.grace.total_seconds(), 60)
+        self.assertEqual(check.last_ping, None)
+        self.assertEqual(check.n_pings, 0)
 
     def test_it_accepts_api_key_in_header(self):
         payload = json.dumps({"name": "Foo"})
 
         ### Make the post request and get the response
-        r = {'status_code': 201} ### This is just a placeholder variable
-
-        self.assertEqual(r['status_code'], 201)
+        r = self.post({"api_key": "abc"})
+        ### r = {'status_code': 201} ### This is just a placeholder variable
+        self.assertEqual(r.status_code, 201)
 
     def test_it_handles_missing_request_body(self):
         ### Make the post request with a missing body and get the response
-        r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
-        self.assertEqual(r['status_code'], 400)
-        self.assertEqual(r["error"], "wrong api_key")
+        ### r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
+        ### self.assertEqual(r['status_code'], 400)
+        ### self.assertEqual(r["error"], "wrong api_key")
+
+        r = self.post({})
+        self.assertEqual(r.status_code, 400)
+        self.assertEqual(r.json()['error'], 'wrong api_key')
```

Comment:
pass the error through the post method above, you have duplicated an assert that is being done there already